openHarmony/openHarmony_dialog.js

```diff
@@ -279,8 +279,12 @@ $.oDialog.prototype.browseForFile = function( text, filter, getExisting, acceptM
     var _files = QFileDialog.getSaveFileName(0, text, startDirectory, filter);
   }
 
-  for (var i in _files){
-    _files[i] = _files[i].replace(/\\/g, "/");
+  if (Array.isArray(_files)) {
+    for (var i = 0; i < _files.length; i++) {
+      _files[i] = new oFile(_files[i].replace(/\\/g, "/"));
+    }
+  } else {
+    _files = new oFile(_files.replace(/\\/g, "/"));
   }
```

Comment:
```suggestion
  try {
    _files = _files.map(function(x){return new $.oFile(x)}; // replace is done during oFile init so not necessary here
  } except {
    // no "map" method means not an array
    _files = new $.oFile(_files);
  }
  ```

Reply:
Yeah I thought it would make problems for backwards compatibility, hmm. It's pretty simple to do the conversion outside the func, obviously, but I think I've been caught out by this a few times before it sunk in. Because most oH functions that ask for files or folders accept oFolder/oFile objects or string paths, it only comes up when you try to use one of the methods.
Do you mean like, some funcs called like chooseFile() and chooseFolder() to distinguish them, and then maybe mark the old ones as deprecated? Another option would be to add an extra bool arg that defaults to false, but that sounds messy to me. Thanks for you comments! I'm learning lots

Reply:
Yes I agree that it would be more consistent if those functions used these classes! I think new functions+a deprecation in a few versions could work, 

Now we have to come up with a good name for it :) browseForFile and browseForFolder were similar to extendScript syntax which is why they were chosen in the first place, so it's a bit of a shame to no longer be able to use them...

Reply:
I realised there was a problem with my implementation too, we need to check if the wrapped file dialog functions return undefined strings to know if the user cancelled the dialog. The way I added the ofile conversion in it was returning ofiles with undefined paths.

It’s tricky choosing the right names for functions! AppleScript which I have used a little uses chooseFile and chooseFolder, which are ok options I think.

I needed to roll back the change I made in my local repo and I’m not sure if I did it in a way which will affect this pull request. I need to learn more about how git works

Reply:
I think chooseFile/chooseFolder can totally be used, we'll keep the old ones for compatibility and people can choose which one they want to use depending.

For adding functions like that, it's a bit special;  you can add them as method of the oDialog class, and also add a general shortcut on the $ object in the openHarmony.js file directly, like it was done for browseForFile/browseForFolder ?

https://github.com/cfourney/OpenHarmony/blob/master/openHarmony/openHarmony_dialog.js#L260

https://github.com/cfourney/OpenHarmony/blob/master/openHarmony.js#L297